src/IO.cpp

```diff
@@ -240,22 +208,64 @@ ObservationsForInference ReadInferenceObservations(const std::string& configDir,
     return observations;
 }
 
+PredictionConfig ReadPredictionConfig(const std::string& configDir, int index, Utilities::logging_stream::Sptr log, const CommonModelInputParameters& commonParameters)
```

Comment:
The difference is that the "Local" function always occurs with or without the data pipeline (there is a call to it from both) since these values are always read from the local files. ReadPredictionConfig only gets called in the non data pipeline, with the equivalent IOdatapipeline::ReadPredictionConfig called for the data pipeline cases.

I just wanted to avoid some duplication. This is similar for the LocalInference function.